src/main/java/com/es/phoneshop/web/ProductListPageServlet.java

```diff
@@ -21,8 +30,22 @@ public void init(ServletConfig config) throws ServletException {
 
     @Override
     protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
-        request.setAttribute("products", productService.findProducts());
-        request.getRequestDispatcher("/WEB-INF/pages/productList.jsp").forward(request, response);
+        String query = Optional.ofNullable(request.getParameter(REQUEST_PARAM_QUERY))
+                .orElse("");
+        ProductSortCriteriaDTO sortCriteria = getSortCriteria(request);
+
+        request.setAttribute(REQUEST_ATTRIBUTE_PRODUCTS, productService.findProducts(query, sortCriteria));
+        request.getRequestDispatcher(REQUEST_DISPATCHER_PRODUCTS).forward(request, response);
+    }
+
+    private ProductSortCriteriaDTO getSortCriteria(HttpServletRequest request) {
+        String sortField = request.getParameter(REQUEST_PARAM_SORT);
+        String sortOrder = request.getParameter(REQUEST_PARAM_ORDER);
+
+        return sortField != null ? ProductSortCriteriaDTO.builder()
+                .sortField(SortField.valueOf(sortField))
+                .sortOrder(SortOrder.valueOf(sortOrder))
+                .build() : null;
```

Comment:
maybe it is better to use Optional here? smth like

```java 
 return Optional.ofNullable(sortField)
            .map(field -> ProductSortCriteriaDTO.builder()
                    .sortField(SortField.valueOf(field))
                    .sortOrder(SortOrder.valueOf(sortOrder))
                    .build()
);
```

end than in next layers write 

```java
Optional.isPresent()
```